### ISSUE TITLE
Ability to locate elements using accessibility values

### ISSUE THREAD

**Issue by codeslord** [SeleniumHQ/selenium#9918]
## 🚀 Feature Proposal

With selenium 4, we should be able to locate elements using their accessibiliy attributes(ARIA) https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA mainly accessibility NAME

## Motivation

I was trying to find the search box with the keyword search in google. After trying it in multiple ways, with relative locators etc I was not able to figure out a way to locate it using the keyword search eventhough it is available (refer to the below screenshot)

![google_accessibility](https://user-images.githubusercontent.com/4618356/137270936-ab2ad5d8-b662-4900-aaa1-67f488370c14.png)

## Example
Something like this
```python
from selenium import webdriver
from selenium.webdriver.support.relative_locator import locate_with
search_element = driver.find_elements(locate_with(By.ACCESSIBILITY, 'Search'))

```

**Comment by AutomatedTester** [comment]
I agree that this is a great feature to have unfortunately it requires the 2 features below to be implemented by browser vendors as they would turn on the accessibility tree in the browser. Another way that we could do it when implemented is when the W3C WICG AOM spec has been shipped a bit more.

https://wpt.fyi/results/webdriver/tests/get_computed_label
https://wpt.fyi/results/webdriver/tests/get_computed_role

**Comment by crunchy-vonage** [comment]
Window.getComputedAccessibleNode(element) is now experimental in chrome, requires passing CLI flags.
See https://wicg.github.io/aom/caniuse.html for details

ChromeDevTools protocol also has direct support in format aria\role[name].
Note they ran benchmarks and decided doing this client side is too slow. So you might need more than get_computed_role as you would need to iterate though every element.
See: https://developer.chrome.com/blog/puppetaria/

I'd be interested in getting some Hacks/POC's running though, it's obviously going to take a long time to get browser support standardised so it would be able to experiment with proposed syntax and APIs.

**Comment by titusfortner** [comment]
I'm going to close this in Selenium tracker. Essentially once it gets added to the spec, we'll implement it in Selenium, and the spec is tracking it here: https://github.com/w3c/webdriver/issues/1441

Like David mentioned, it's all held up by the browser vendors not yet having provided access to the tree in the first place.
